### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by olivierlemasle**
- Upgrade k8s dependencies to 0.25
- Upgrade github.com/emicklei/go-restful to v3

NB: `k8s.io/apiserver/pkg/registry/rest.Storage` interface now has a new `Destroy()` function.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,18 +3,18 @@ module sigs.k8s.io/custom-metrics-apiserver
 go 1.18
```
Comment (dgrisonnet) [inline go.mod:3]: Let's require `go 1.19` here since it is the case for k8s 0.25.4 deps: https://github.com/kubernetes/client-go/blob/v0.25.4/go.mod

**Comment by olivierlemasle** [comment]
@dgrisonnet Updated to use go 1.19

Cf https://github.com/kubernetes/test-infra/pull/28117 for updating CI.

**Comment by dgrisonnet** [comment]
Thanks for creating the test-infra PR! :)

**Pull request merged**